src/puppetlabs/kitchensink/core.clj

```diff
@@ -260,6 +259,47 @@ to be a zipper."
       (last weights-and-values)
       (-> (first selected) second))))
 
+(def ascii-character-sets
+  (let [ALPHA (map char (range 65 91))
+        alpha (map char (range 97 123))
+        digits (map char (range 48 58))
+        symbols (concat (map char (range 33 48))
+                        (map char (range 91 97))
+                        (map char (range 123 127)))]
+    {:alpha (concat alpha ALPHA)
+     :alpha-lower alpha
+     :alpha-upper ALPHA
+     :alpha-digits (concat alpha ALPHA digits)
+     :alpha-digits-symbols (concat alpha ALPHA digits symbols)
+     :symbols symbols
+     :digits digits}))
+
+(defn rand-str
+  "Produces a random string of length n, drawn from the given collection of
+  characters. The following keywords may be used in place of a character
+  collection:
+    :alpha - [a-zA-Z]
+    :alpha-lower - [a-z]
+    :alpha-upper - [A-Z]
+    :alpha-digits - [a-zA-Z0-9]
+    :alpha-digits-symbols - all printable ASCII characters besides space
+    :symbols - all visible, non-alpha-numeric ASCII characters (no space)
+    :digits - [0-9]
+  If no character collection or keyword is provided, :alpha-digits-symbols is
+  used by default."
+  ([n] (rand-str :alpha-digits-symbols n))
+  ([characters n]
+   (let [char-coll (cond
+                     (and (keyword? characters) (contains? ascii-character-sets characters))
+                     (get ascii-character-sets characters)
+
+                     (keyword? characters)
```

Comment:
Maybe could validate this as a function pre-condition like we've done on other functions in this namespace rather than down at this level.  This seems fine as well, though.